dnstable/entry.c

```diff
@@ -79,32 +79,45 @@ fmt_time(ubuf *u, uint64_t v)
 		ubuf_add_cstr(u, s);
 }
 
+static inline void
+fmt_rfc3339(const struct tm *src, char *dst)
+{
+	my_uint64_to_str_padded(1900 + src->tm_year, 4, dst);
+	my_uint64_to_str_padded(1 + src->tm_mon, 2, dst + 5);
+	my_uint64_to_str_padded(src->tm_mday, 2, dst + 8);
+	my_uint64_to_str_padded(src->tm_hour, 2, dst + 11);
+	my_uint64_to_str_padded(src->tm_min, 2, dst + 14);
+	my_uint64_to_str_padded(src->tm_sec, 2, dst + 17);
+}
+
 static void
 fmt_rfc3339_time(ubuf *u, uint64_t v)
 {
 	struct tm gm, *r;
 	time_t tm = v;
-	char s[sizeof("4294967295-12-31T23:59:59Z")];
+	char ts[]="2023-12-31T23:59:59Z";
 
 	r = gmtime_r(&tm, &gm);
-	if ((r != NULL) && (strftime(s, sizeof(s), "%Y-%m-%dT%H:%M:%SZ", r) > 0))
-		ubuf_add_cstr(u, s);
+	assert(r != NULL);
+
+	fmt_rfc3339(r, ts);
+
+	ubuf_append_cstr_lit(u, ts);
```

Comment:
`ts` is an array, not a string literal as required by `ubuf_append_cstr_lit`.

Reply:
Pardon my confusion: `sizeof` works fine on array types.

A comment to the effect of:
```
fmt_rfc3339 will completely fill the ts array, so sizeof can be used to compute the string length
```
may be useful clarification.